ironfish/src/indexers/minedBlocksIndexer.ts

```diff
@@ -0,0 +1,369 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { Accounts } from '../account'
+import { Assert } from '../assert'
+import { Blockchain } from '../blockchain'
+import { ChainProcessor } from '../chainProcessor'
+import { FileSystem } from '../fileSystems'
+import { createRootLogger, Logger } from '../logger'
+import {
+  ArrayEncoding,
+  BufferEncoding,
+  IDatabase,
+  IDatabaseStore,
+  IDatabaseTransaction,
+  JsonEncoding,
+  StringEncoding,
+  U32Encoding,
+} from '../storage'
+import { createDB } from '../storage/utils'
+import { SetTimeoutToken } from '../utils'
+import { BlockchainUtils, isBlockMine } from '../utils/blockchain'
+
+const DATABASE_VERSION = 1
+const REMOVAL_KEY = 'accountsToRemove'
+
+const getMinedBlocksDBMetaDefaults = (): MinedBlocksDBMeta => ({
+  headHash: null,
+})
+
+type MinedBlocksDBMeta = {
+  headHash: string | null
+}
+
+type MinedBlock = {
+  main: boolean
+  sequence: number
+  account: string
+  minersFee: number
+}
+
+export class MinedBlocksIndexer {
+  protected meta: IDatabaseStore<{
+    key: keyof MinedBlocksDBMeta
+    value: MinedBlocksDBMeta[keyof MinedBlocksDBMeta]
+  }>
+  protected minedBlocks: IDatabaseStore<{ key: Buffer; value: MinedBlock }>
+  protected sequenceToHashes: IDatabaseStore<{ key: number; value: Buffer[] }>
+  protected accountsToRemove: IDatabaseStore<{ key: string; value: string[] }>
+
+  protected files: FileSystem
+  protected database: IDatabase
+  protected location: string
+  protected readonly accounts: Accounts
+  protected readonly logger: Logger
+  protected isOpen: boolean
+  protected isStarted: boolean
+  protected rescan: boolean
+  protected eventLoopTimeout: SetTimeoutToken | null = null
+  protected chain: Blockchain
+  protected chainProcessor: ChainProcessor
+
+  constructor({
+    files,
+    location,
+    accounts,
+    chain,
+    logger = createRootLogger(),
+  }: {
+    files: FileSystem
+    location: string
+    accounts: Accounts
+    chain: Blockchain
+    logger?: Logger
+  }) {
+    this.files = files
+    this.location = location
+    this.database = createDB({ location })
+    this.accounts = accounts
+    this.logger = logger
+    this.chain = chain
+    this.isOpen = false
+    this.isStarted = false
+    this.rescan = false
+
+    this.meta = this.database.addStore<{
+      key: keyof MinedBlocksDBMeta
+      value: MinedBlocksDBMeta[keyof MinedBlocksDBMeta]
+    }>({
+      name: 'meta',
+      keyEncoding: new StringEncoding<keyof MinedBlocksDBMeta>(),
+      valueEncoding: new JsonEncoding(),
```

Comment:
We now have binary encodings for all our stores in the node. Can you add encoders for each of these stores? You can look at the [PR](https://github.com/iron-fish/ironfish/pull/1256) here for examples. [Here](https://github.com/iron-fish/ironfish/blob/master/ironfish/src/account/database/accounts.ts) is how accounts are stored.

It might be useful to do these as separate PRs into this branch.